File: easy/1320.ContainsDuplicate/s.go
/**
 * @param nums: the given array
 * @return: if any value appears at least twice in the array
 */
func containsDuplicate(nums []int) bool {
	// Write your code here
	seen := make(map[int]bool)
	for _, num := range nums {
		if _, ok := seen[num]; ok {
			return true
		} else {
			seen[num] = true
		}
	}
	return false
}

File: easy/1320.ContainsDuplicate/s.py
class Solution:
    """
    @param nums: the given array
    @return: if any value appears at least twice in the array
    """
    def containsDuplicate(self, nums):
        # Write your code here
        return len(set(nums)) != len(nums)
       
    def containsDuplicate2(self, nums):
        # Write your code here
        # use set
        seen = set()
        for num in nums:
            if num in seen:
                return True
            else:
                seen.add(num)
        return False